fcrepo-clients/src/test/java/edu/unc/lib/dl/event/RepositoryPremisLoggerIT.java

```diff
@@ -166,4 +181,221 @@ public void getEventsModelForObjectWithoutLog() throws Exception {
+        assertTrue(logEvent1Resc.hasProperty(RDF.type, Premis.VirusCheck));
+
+        Resource logEvent2Resc = logModel.getResource(event2Resc.getURI());
+        assertTrue(logEvent2Resc.hasProperty(RDF.type, Premis.Ingestion));
+        assertEquals("2010-01-02T12:00:00.000Z", logEvent2Resc.getProperty(DCTerms.date).getString());
+
+        int i = 1;
+        for (String uri : eventUris) {
+            Resource logEventResc = logModel.getResource(uri);
+            assertEquals("another premis event " + i, logEventResc.getProperty(Premis.note).getString());
+            i++;
+        }
+    }
+
+    @Test
+    public void allowSimultaneousReadLocks() throws InterruptedException {
+        parentObject = repoObjFactory.createDepositRecord(null);
+        initPremisLogger(parentObject);
+
+        // make sure that there are no events in premis log before the writes
+        PremisLogger retrieveLogger = new RepositoryPremisLogger(parentObject, mockSession,
+                pidMinter, repoObjLoader, repoObjFactory);
+        Model initialLogModel = retrieveLogger.getEventsModel();
+        assertFalse("New premis already contains events", initialLogModel.listObjects().hasNext());
+
+        // add an event
+        Resource event = logger.buildEvent(Premis.note)
+                .addEventDetail("first premis event")
+                .write();
+
+        // create a read lock
+        PID logPid = DatastreamPids.getMdEventsPid(parentObject.getPid());
+        Lock logLock = lockManager.awaitReadLock(logPid);
+
+        // start a thread to read the log
+        List<String> premisNotes = new ArrayList<>();
+        Runnable readThreadRunnable = new Runnable() {
+            @Override
+            public void run() {
+                Model logModel = retrieveLogger.getEventsModel();
+                Resource logEventResc = logModel.getResource(event.getURI());
+                premisNotes.add(logEventResc.getProperty(Premis.note).getString());
+            }
+        };
+        Thread readThread = new Thread(readThreadRunnable);
+        readThread.start();
+        readThread.join();
+        assertEquals("first premis event", premisNotes.get(0));
+
+        // release read lock
+        logLock.unlock();
+    }
+
+    @Test
+    public void readLockWaitsForWriteLock() throws InterruptedException {
+        parentObject = repoObjFactory.createDepositRecord(null);
+        initPremisLogger(parentObject);
+
+        // make sure that there are no events in premis log before the writes
+        PremisLogger retrieveLogger = new RepositoryPremisLogger(parentObject, mockSession,
+                pidMinter, repoObjLoader, repoObjFactory);
+        Model initialLogModel = retrieveLogger.getEventsModel();
+        assertFalse("New premis already contains events", initialLogModel.listObjects().hasNext());
+
+        // add event
+        Resource event = logger.buildEvent(Premis.note)
```

Comment:
since the test isn't ever checking for the presence of this event, seems like we could probably remove it